### ISSUE TITLE
Add ens.etsmtl.ca email authentication

### ISSUE THREAD

**Issue by robinmarce** [ApplETS/Notre-Dame#390]
The authentication environment of ETS is mostly email adress + password. I think it is confusing to authenticate with the universal code which I didn't know by heart when I arrived at ETS. I had to look for it to have access to the app.
Also, for password managers it is confusing to have several logins for the same password.
This morning, when showing the app to a classmate, my classmate tried to sign in with his email adress.
We can see that as a careless mistake but we can also see that as a possible change for a not instinctive behaviour of the app.

I would like to be able to sign in with my ETS email adress. Maybe we could enable both email adress and universal code like Google's authentication (email or phone number).

**Comment by apomalyn** [comment]
Not that we don't want to do it but currently we don't have the API endpoint to do that.

This could be discussed with the school.

**Comment by MysticFragilist** [comment]
Hi!

That's a great point and i think it would definitely be beneficial for sure! :smile:

Sadly, as of right now, the API we used to access resources (course, schedule, grade, etc) is provided as is from the TI department of the school. It's a difficult one, because we could bubble up the issue to them, but it's really not garanteed they will be onboard about changing the entire authentication system. 

We have to keep in mind the auth system is based on Signets old way of connecting to your portal and is needed for **every** API calls. So i have to believe that for them, it would be kind of difficult to change that without changing the entire architecture.

Apart from this hiccup, I found this to be a great idea for newcomers! As I was myself a bit confused when launching for the first time the app. :smile:

This is just food for thought, I'll put this issue on blocked for now and if enough person needs this or find this useful, let me know and we will return to this, no problem!

Also, @apomalyn you told me that there was the other API from MonETS that was supposed to be built with more modern functionalities right? But was abandoned or put on hold for some time now?

-Sam

**Comment by apomalyn** [comment]
Exactly, the work on MonETSApi is still in progress (MonETS portal consume it I think) but we don't have any documentation or information about it...you could try to ask again for it maybe this time we will have an update 😅

**Comment by MysticFragilist** [comment]
I'll look into that with Christian!

**Comment by XavierPaquet-Rapold** [comment]
Replaced by #1143
